### ISSUE TITLE
Bug: third-gen visibility threshold doesn't seem to do anything

### ISSUE THREAD

**Issue by oneillkza** [igvteam/igv#736]
A common use case I have is having long-read sequence data as well as short reads, and wanting to examine fairly large structural variants. (e.g. I'm currently working with a 400Kb duplication). 

I do still want the short read data loaded in the session, since I'm having to zoom in to look at the breakpoints, but at the same time I want to be able to see the Nanopore reads when I'm zoomed out to a ~500Kb window.

To do this, I've set the visibility range for the third gen reads to 1Mb, and set the alignment visibility range to 50Kb.

However, IGV only seems to be respecting the 50Kb setting.

Is it possible it isn't recognising the Nanopore bam as a third gen track? Is there a way to force a track to be treated as third gen?

**Comment by jrobinso** [comment]
I noticed this myself, coincidentally, and opened this issue for it:
https://github.com/igvteam/igv/issues/732

In brief, before any alignments are read the experiment type is not known,
so it defaults to the base defaults.  After zooming in to load alignments
it should detect it as 3rd gen.

**Comment by jrobinso** [comment]
@oneillkza Can you extract, zip, and attach a small nanopore bam file, or point me to a public nanopore bam file?

**Comment by jrobinso** [comment]
btw,  #732 is part of the issue here, but there are others.  Going through it now.

**Comment by jrobinso** [comment]
@oneillkza I've pushed a fix for this, available in the snapshot build.   The type of the experiment will remain unknown until alignments are loaded, but there is now a menu item to explicitly set it in the popup.   I could not test Nanopore data as I don't have any available, if you are able to contribute or point me to a test dataset that would be helpful.

**Comment by oneillkza** [comment]
Here's a small Nanopore bam file, generated from public NA19240 data, only including reads overlapping 21:10000000-1010000.

https://www.bcgsc.ca/downloads/koneill/na19240_subset.bam
https://www.bcgsc.ca/downloads/koneill/na19240_subset.bam.bai

It was aligned using minimap2, which seems to be coalescing as the standard aligner for Nanopore, but there are others. In some ways it might be best just to peek at the first few reads and if they're all different lengths and quite long, then to consider the data to be third-gen.

**Comment by oneillkza** [comment]
I just tested the snapshot with both Illumina and Nanopore bams, and it seemed to perform correctly, respecting both viewing windows. Thanks!

**Comment by jrobinso** [comment]
@oneillkza Thanks for the test file, its the only platform I had nothing for.  RE detection that's essentially what I do,  but only in the process of zooming into an actual region, which of course is too late to set an initial visibility window.   I worried a bit about pulling reads from the front of the file, that they could be somehow atypical, but that's probably nothing to worry about.  In any event that's what #732 is about so I'll close this one.

**Comment by jrobinso** [comment]
@oneillkza BTW, for the record, here is the test.  Suggestions for improvements welcome.  readLenthStdDev is the standard deviation.

```
       if (readStats.readLengthStdDev > 100 || readStats.medianReadLength > 1000) {
            setExperimentType(AlignmentTrack.ExperimentType.THIRD_GEN);  // Could also use fracReadsWithIndels
        } else if (readStats.medianRefToReadRatio > 10) {
            setExperimentType(AlignmentTrack.ExperimentType.RNA);
        } else {
            setExperimentType(AlignmentTrack.ExperimentType.OTHER);
        }
```

**Comment by oneillkza** [comment]
That looks pretty good. The only case I could see it missing would be long-read amplicons shorter than 1Kb, but in that case folks can set the type manually.

I'd also note that I've loaded both cDNA and direct-sequenced RNA data in IGV and it seems to automatically detect it as RNA, which displays sensible things (even for the long reads).

**Comment by helgathorv** [comment]
'Experiment type' in alignment track menu - released in IGV 2.8.0
